src/nti/contentfragments/html.py

```diff
@@ -195,6 +196,7 @@ def __iter__(self):
     def _find_links_in_text(self, token):
         text = token['data']
         text_and_links = self.link_finder.find_links(text) if self.link_finder else ()
+        #if len(text_and_links) == 0 we return None instead of token
```

Comment:
Thoughts on this block? If text_and_links is empty, we enter this block and return `None`. I was not sure if we wanted to return the token by default in this case.

Reply:
This is a generator, so I think what happens in that case is that there's simply nothing to iterate over; the generator yields no items.